sanic/cookies/response.py

```diff
@@ -389,32 +392,66 @@ def delete_cookie(
             This requires that secure=True, defaults to False
         :type secure_prefix: bool
         """
+
+        # Host prefix can only be used on a cookie where
+        # path=="/", domain==None, and secure==True
+        host_prefix = (
+            False
+            if not (secure and path == "/" and domain is None)
+            else host_prefix
+        )
+        # Secure prefix can only be used on a secure cookie
+        secure_prefix = False if not secure else secure_prefix
```

Comment:
Hiding safety rails leads to hard to debug code. Better off failing explicitly with an error.